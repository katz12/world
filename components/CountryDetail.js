import React from 'react';
import { Container, Header } from 'semantic-ui-react';

import QueryViewer from './QueryViewer';
import { getPopulationQuery, countryCodeMap } from '../api/dbpedia';

const CountryDetail = ({ selectedCountryCode }) => {
  if (!selectedCountryCode) {
    return null;
  } else {
    return (
      <Container>
        <Header as='h2' dividing style={{marginTop: '1em'}}>
          Querying data from: {countryCodeMap[selectedCountryCode]}
        </Header>
        <QueryViewer key={selectedCountryCode} initialQuery={getPopulationQuery(selectedCountryCode)} />
      </Container>
    )
  }
}

export default CountryDetail;
