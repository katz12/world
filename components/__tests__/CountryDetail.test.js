import React from 'react';
import CountryDetail from '../CountryDetail';
import { shallow } from 'enzyme';

it('should render with props', () => {
  expect(shallow(<CountryDetail selectedCountryCode="US" />).debug()).toMatchSnapshot()
})

it ('should not render without props', () => {
  expect(shallow(<CountryDetail />).debug()).toMatchSnapshot()
})
