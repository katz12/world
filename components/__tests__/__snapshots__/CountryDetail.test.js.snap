// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render without props 1`] = `""`;

exports[`should render with props 1`] = `
"<Container>
  <Header as=\\"h2\\" dividing={true} style={{...}}>
    Querying data from: 
    http://dbpedia.org/resource/United_States
  </Header>
  <QueryViewer initialQuery=\\"SELECT * WHERE {\\\\n\\\\tOPTIONAL {<http://dbpedia.org/resource/United_States> dbo:populationTotal ?populationTotal.}\\\\n}\\" />
</Container>"
`;
